gemrb/core/Scriptable/Actor.cpp

```diff
@@ -864,9 +864,11 @@ static void UpdateHappiness(Actor *actor) {
 	if (!core->HasFeature(GFFlags::HAPPINESS)) return;
 
 	ieWordSigned newHappiness = GetHappiness(actor, core->GetGame()->Reputation);
-	if (newHappiness == actor->PCStats->Happiness) return;
+	if (actor->PCStats) {
```

Comment:
We can just return  here if it's not available, since the rest of the function doesn't make sense without the PCStats block.

Reply:
Ok.

Reply:
Done rework it